src/main/java/duke/task/TaskBank.java

```diff
@@ -0,0 +1,148 @@
+    }
+
+    // search the Task in the array by index
+    public Task searchTask(int taskIndex) throws IndexOutOfBoundsException {
+        if (taskIndex < 0 || taskIndex > this.getTaskSize()) {
+            throw new IndexOutOfBoundsException("Ouch! Index is out of range. Try again!\n");
+        }
+        return tasks.get(taskIndex);
+    }
+
+    public Task removeTask(int taskIndex) throws IndexOutOfBoundsException {
+        if (taskIndex < 0 || taskIndex > this.getTaskSize()) {
+            throw new IndexOutOfBoundsException("Ouch! Index is out of range. Try again!\n");
+        }
+        Task deletedTask = tasks.get(taskIndex);
+        tasks.remove(taskIndex);
+        return deletedTask;
+    }
+
+    public void exportTasks() {
+        StringBuffer taskTextString = new StringBuffer();
+        for (Task task : tasks) {
+            taskTextString.append(task.getTaskType());
+            taskTextString.append(" | ");
+            taskTextString.append(task.getDone() ? "1" : "0");
+            taskTextString.append(" | ");
+            taskTextString.append(task.describe());
+            taskTextString.append("\r\n");
+        }
+        try {
+            FileWriter fw = new FileWriter(filePath);
+            fw.write(taskTextString.toString());
+            fw.close();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public void loadTasks() {
+        File f = new File(filePath);
+        try (Scanner sc = new Scanner(f)) {
+            while (sc.hasNext()) {
+                loadTaskLine(sc.nextLine());
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("duke.txt is not found");
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+
+    }
+
+    public void loadTaskLine(String taskLine) throws IOException {
+        String taskTypeString = taskLine.substring(0, 1);
+        int firstDivisor = taskLine.indexOf("|", 1);
+        int secondDivisor = taskLine.indexOf("|", firstDivisor + 1);
+        int thirdDivisor = taskLine.indexOf("|", secondDivisor + 1);
+        if (thirdDivisor == -1) { // todo type
+            addTodo(taskLine.substring(secondDivisor + 1).trim());
+        } else {
+            if (taskTypeString.equals("D")) {
+                String deadLineInput = taskLine.substring(secondDivisor + 1).trim();
+                String taskCompletionStatus = taskLine.substring(firstDivisor + 2, secondDivisor).trim();
+                Task newTask = addDeadline(deadLineInput.replace("| ", "/"));
+                if (taskCompletionStatus.equals("1")) {
+                    try {
+                        newTask.markAsDone();
+                    } catch (RepeatedCompletionException e) {
+                        // This is left blank intentinally
+                        // as from tasks are generated from
+                        // local files, and will not be completed repeatedly
```

Comment:
You are recommended to handle exceptions in catch block even though you expect it to not be raised (there could be instances where your code consists of bugs that caused this exception so it is better for you to handle it as well).

Reply:
What should be the correct way to handle this exception? I should neither print anything nor raise the exception. This is a self-defined exception that could only be thrown at a specific instance.